### DISCUSSION TITLE
Restore after a crash

### DISCUSSION THREAD

**Discussion by tou-louse** [prodrigestivill/docker-postgres-backup-local#148] [Q&A]
how can i restore a backup in unraid with postgres-backup-local, the backups are on an external hard disk which is embedded in the container. The command from the instructions does not work. I always get the message that it is not a container

i using this command

docker exec --tty --interactive $postgresql14 /bin/sh -c "zcat $postgres-latest.sql.gz | psql --username=$postgres --dbname=$postgre
s -W"

**Comment by prodrigestivill** [comment]
docker exec runs the command in an already running container, do you have the container already running? if so, please use the conainer id instead of `$posgresql14`.

And follow all the requiered substitutions explained in the [README](https://github.com/prodrigestivill/docker-postgres-backup-local?tab=readme-ov-file#restore-examples) file.

**Comment by tou-louse** [comment]
Hi,
when i use the container id, i become the same error message

Error: No such container: /bin/sh

The Container is already running and other Container, there using postgres, how plex, are running correct

**Reply by prodrigestivill** [reply]
did you substitute the ids including the `$`?

**Comment by tou-louse** [comment]
Hi, this is my command

docker exec --tty --interactive **$b422e0afe654** /bin/sh -c "zcat **$postgres-20240229.sql.gz** | psql --username=**$username** --**dbname=$dbname** -W"

**Reply by prodrigestivill** [reply]
try removing all `$` from your command line.

**Comment by tou-louse** [comment]
thx for the tipp...

It now accepts the command, but cannot do anything with the file. The file is in the backup folder that is stored in postgres-Backup-local. Where am I making the mistake?

this is the error message

gzip: postgres-20240229.sql.gz: No such file or directory

thx again

**Reply by prodrigestivill** [reply]
you need to use the full path, to restore the file.

**Reply by prodrigestivill** [reply]
Example: `/backups/daily/DBNAME-20240229.sql.gz`
